### PR TITLE
windows: return bind error immediately when implicitly binding

### PR DESCRIPTION
**Pull request by saghul**
If a TCP handle is bound automatically when calling uv_listen or
uv_connect, there is a remote possibility that it fails, return
the error if so.

### CODE REVIEW

**Comment by saghul** [comment]
Found it while checking the code for something else. /cc @indutny

File: src/win/tcp.c [hunk 1]
```diff
@@ -539,6 +539,8 @@ int uv_tcp_listen(uv_tcp_t* handle, int backlog, uv_connection_cb cb) {
                           0);
     if (err)
       return err;
+    if (handle->flags & UV_HANDLE_BIND_ERROR)
```
Comment (indutny) [inline src/win/tcp.c:542]: Wait, isn't this a success branch?

Comment (saghul) [inline src/win/tcp.c:542]: Yes. `uv_tcp_try_bind` will return 0 in case of EADDRINUSE and set the flag and handle->bind_error to the actual error.

Comment (indutny) [inline src/win/tcp.c:542]: hm... and what if `bind_error` is `0`?

Comment (saghul) [inline src/win/tcp.c:542]: Can't be if `handle->flags & UV_HANDLE_BIND_ERROR` is set.

**Comment by indutny** [comment]
Aaaah, I see. Sorry! LGTM

**Comment by saghul** [comment]
No worries, thanks Fedor!

**Pull request merged**